### PR TITLE
Fix call sites of hipmemset

### PR DESCRIPTION
**Pull request by whchung**
hipMemset() and hipMemsetAsync() do take integer-typed parameters so checks
here make no sense.

Relevant PR in HIP: https://github.com/ROCm-Developer-Tools/HIP/pull/933

### CODE REVIEW

**Comment by whchung** [comment]
@deven-amd we don't need HIP PR 933 be merged to take this PR. On non-XLA path only 0 would ever be passed to `hipMemset` and `hipMemsetAsync`. But for XLA there are tests to use non-0 init values and we'll need HIP PR 933 then.

File: tensorflow/stream_executor/rocm/rocm_driver.cc [hunk 1]
```diff
@@ -486,15 +486,8 @@ GpuDriver::ContextGetSharedMemConfig(GpuContext* context) {
                                                      size_t uint32_count) {
   ScopedActivateContext activation{context};
   void* pointer = absl::bit_cast<void*>(location);
-  unsigned char valueC = static_cast<unsigned char>(value);
-  uint32_t value32 = (valueC << 24) | (valueC << 16) | (valueC << 8) | (valueC);
```
Comment (deven-amd) [inline tensorflow/stream_executor/rocm/rocm_driver.cc]: wont this logic now be needed in the call to hipMemset in the SynchronousMemsetUint8 routine (line 450 above)?

File: tensorflow/stream_executor/rocm/rocm_driver.cc [hunk 2]
```diff
@@ -524,15 +517,6 @@ GpuDriver::ContextGetSharedMemConfig(GpuContext* context) {
                                                       GpuStreamHandle stream) {
   ScopedActivateContext activation{context};
   void* pointer = absl::bit_cast<void*>(location);
-
-  // FIXME - need to set a 32-bit value here
-  unsigned char valueC = static_cast<unsigned char>(value);
-  uint32_t value32 = (valueC << 24) | (valueC << 16) | (valueC << 8) | (valueC);
```
Comment (deven-amd) [inline tensorflow/stream_executor/rocm/rocm_driver.cc]: same comment as before...do we need to add this logic for the call to hipMemsetAsync at line 511

File: tensorflow/stream_executor/rocm/rocm_driver.cc (hunk 1 above)
Comment (whchung) [reply tensorflow/stream_executor/rocm/rocm_driver.cc]: https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/hcc_detail/hip_runtime_api.h#L1494

this line is correct.

Comment (whchung) [reply tensorflow/stream_executor/rocm/rocm_driver.cc]: and for the call site of U8, probably need to switch to:
https://github.com/ROCm-Developer-Tools/HIP/blob/master/include/hip/hcc_detail/hip_runtime_api.h#L1505

Comment (deven-amd) [reply tensorflow/stream_executor/rocm/rocm_driver.cc]: yes, switching the call at line 450 to hipMemsetD8 would work too.

**Comment by whchung** [comment]
It seems in `//tensorflow/compiler/xla/tests:select_and_scatter_gpu` there are some complicated memset behaviors. I'll study it better to understand how `rocm_driver`, and perhaps HIP APIs, be chagned.

**Comment by whchung** [comment]
@deven-amd could you give this PR another go. In HIP PR ( https://github.com/ROCm-Developer-Tools/HIP/pull/933 ), I added 2 new APIs in HIP so avoid inadvertently changing the semantic of original APIs.

**Comment by deven-amd** [comment]
LGTM

**Comment by whchung** [comment]
HIP PR has landed.

**Comment by deven-amd** [comment]
jenkins: retest this please

**Pull request merged**